jdk398/Shadow_of_Knight_Solution.cpp

```diff
@@ -0,0 +1,163 @@
+example, in the picture above the bomb is far up and to the right, so the next location 
+we would want to choose would be to a location like (10,2). 
+
+Something that also should be noted about the above picture are the two red lines flanking Batman. These are a graphical representation of what I call "limits". Limits are essentially coordinates, either x or y, that Batman is restricted from going past. So in a future move Batman will use the limits defined before as an outer bounds to help calculate his next move. We decide limits based on the direction that the game gives us for the bomb. For our example above, because the bomb is above and to the right, we know for sure that it is not anywhere below or to the left of our current position. by using this method repeatedly we cut down the search area until we can make our final move with certainty. 
+
+Code 
+====
+The following will be the source code for the solution, along with some explanation and reasoning behind the solution process.
+
+Includes
+--------
+*/
+#include <iostream>
+#include <string>
+#include <vector>
+#include <algorithm>
+
+using namespace std;
+
+
+// Provided Code
+// -------------
+// The provided code for the game, these functions and variables have been set up to hand the user the required information from the game to be able to create the solution.
+int main()
+{
+    // The following variables are one time inputs for the game. h and w determine the height and width of the building that Batman is scaling respectively.
+    int w; 
+    int h; 
+    cin >> w >> h; cin.ignore();
+    // n is the maximum number of turns before game over. (we use this very little)
+    int n; 
+    cin >> n; cin.ignore();
+    // x0 and y0 represent the initial coordinates of Batman.
+    int x0;
+    int y0;
+    cin >> x0 >> y0; cin.ignore();
+
+    
+// My Code
+// -------
+    // The location of batman, also count, which is just used to check if we are in the first round, this is for some levels it is important to start in the right position.
+    int cur_batx, cur_baty, count;
+    // Variable Definitions for the limits of where the bomb could be from the top, bottom, left, and right
+    int upr_lim, lwr_lim, lft_lim, rgt_lim;
+    // My final solution used what I called limits (defined above). The basic idea is that if the bomb is in some direction from batman, we should not go in the opposite direction. This only becomes an issue in later stages where overshooting is likely.
+
+    // count is initially set to zero, and the inital limits are set
+    count = 0;
+    // These limits are essentially the edge of the board, because we don't know where the bomb could be.
+    upr_lim = 0;
+    lwr_lim = h;
+    lft_lim = 0;
+    rgt_lim = w;
+
+    // These are the variables that I send to the output to pick batman's next location.
+    int batx = x0;    // Selects the middle square to start from
+    int baty = y0;
+    // 
+    // ---------
```

Comment:
Don't indent headings -- it will mess formatting up. Either use something besides a heading, or un-indent this.